app/Helpers/Bank/Nordigen/Nordigen.php

```diff
@@ -60,22 +60,58 @@ public function getInstitutions()
         return $this->client->institution->getInstitutions();
     }
 
-    // requisition-section
-    public function createRequisition(string $redirect, string $institutionId, string $reference, string $userLanguage)
-    {
-        if ($this->test_mode && $institutionId != $this->sandbox_institutionId) {
+    /**
+     * Get end user agreement details by ID.
+     *
+     * @return array{
+     *   id: string,
+     *   created: string,
+     *   institution_id: string,
+     *   max_historical_days: int,
+     *   access_valid_for_days: int,
+     *   access_scope: string[],
+     *   accepted: string
+     * } Agreement details
+     */
+    public function getAgreement(string $euaId): array {
+        $eua = $this->client->endUserAgreement->getEndUserAgreement($euaId);
+
+        return $eua;
+    }
```

Comment:
i have included this method within the PR #10428 and other changes as well, f.ex. to reduce creating of unnecessary new endUserAgreements

please wait until its merged to v5-develop and migrate to the new Nordigen class implementation within your other changes